### DISCUSSION TITLE
I've made a template project for developing custom widgets

### DISCUSSION THREAD

**Discussion by iamspark1e** [decaporg/decap-cms#6874] [Show and tell]
Dear all,

My template project: [iamspark1e/decap-cms-widget-starter](https://github.com/iamspark1e/decap-cms-widget-starter)

While developing plugins for decap-cms, I found that it can be a bit daunting to follow the instructions in the "Creating Custom Widgets" chapter. Additionally, the Webpack and related plugins that are referenced in that chapter are no longer the latest versions.

Therefore, I have created a template project for developing custom plugins, using Vite as bundler, and works perfectly both in "dev" and "production". I hope that this will be helpful to other developers.

( I would like to add my template project to our documentation. However, I am not sure whether it should be added to the "Creating Custom Widgets" chapter or the "Examples" chapter. 😕  )

**Comment by martinjagodic** [comment]
@iamspark1e this is great! Before we add it to official docs, could you document your project and write a more detailed readme?

**Reply by iamspark1e** [reply]
Sure, I'd be happy to do that. I'll update this discussion once I'm done adding the document & readme 😃 .

**Comment by iamspark1e** [comment]
@martinjagodic Some guidence for my starter template is added in README now 😄
